docs/developer-guide/join-streams-and-tables.rst

```diff
@@ -0,0 +1,329 @@
+.. _join-streams-and-tables:
+
+Join Streams and Tables with KSQL
+=================================
+
+You can use KSQL to merge streams of data in realtime by using a SQL-like
+*join* syntax. A join in KSQL is similar to a join clause in a relational
+database, which combines data from two sources based on values common to
+both. The result of a KSQL join is a new stream or table that's populated
+with the column values that you specify in a SELECT statement.
+
+KSQL automates creating Kafka Streams applications. You don’t need to write
+the low-level logic around joining streams, so you can focus on the business
+logic for combining your streaming data.
+
+You can join streams and tables in three different ways:
+
+* Join two streams to create a new stream.
+* Join two tables to create a new table.
+* Join a stream and a table to create a new stream.
+
+JOIN Clause
+***********
+
+The KSQL JOIN clause has the familiar syntax of a SQL JOIN clause.
+The following example creates a ``pageviews_enriched`` stream, which is a
+combination of a ``pageviews`` stream and a ``users`` table:
+
+.. code:: sql
+
+   CREATE STREAM pageviews_enriched AS
+     SELECT users.userid AS userid, pageid, regionid, gender FROM pageviews
+     LEFT JOIN users ON pageviews.userid = users.userid;
+
+
+For the full code example, see :ref:`ksql_quickstart-docker`.
+
+Here's an example stream-stream join that combines a ``shipments`` stream with
+an ``orders`` stream, within a time window. The resulting ``late_orders`` stream
+detects late orders by matching each shipments row with all orders rows that
+occur within a two-hour window. If there's no match, the right-hand side of the
+join result is NULL, which indicates that the order wasn't shipped within the
+expected time of two hours.
+
+.. code:: sql
+
+   CREATE STREAM late_orders AS
+     SELECT o.orderid, o.itemid FROM orders o
+     FULL OUTER JOIN shipments s WITHIN 2 HOURS
+     ON s.orderid = o.orderid WHERE s.orderid IS NULL;
+
+For more information, see Join Syntax in the KSQL Syntax Reference (TBD).
+
+Joins and Windows
+*****************
+
+KSQL enables grouping records that have the same key for stateful operations,
+like joins, into *windows*. You specify a retention period for the window, and

```

Comment:
We should call out that only stream-stream joins are windowed. A link to the config which controls window retention will also be helpful in this context.